doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,228 @@
+     - ``"forest.qvm"``, ``qvm_url`` needs to be set to use the Lisp QVM
+     - Forest QVM device supporting both the Rigetti Lisp QVM, as well as the built-in pyQuil pyQVM
+   * - `AQT Ideal ion-trap simulator <https://pennylane-aqt.readthedocs.io/en/latest/devices.html#ideal-ion-trap-simulator>`__
+     - ``"aqt.sim"``
+     - Ideal noiseless ion-trap simulator
+   * - `AQT Noisy ion-trap simulator <https://pennylane-aqt.readthedocs.io/en/latest/devices.html#noisy-ion-trap-simulator>`__
+     - ``"aqt.noisy_sim"``
+     - Noisy ion-trap simulator for more realisatic simulations
+
+:html:`</div></div>`
+
+
+:html:`<div class="hw_list" id="aside1"><a data-toggle="collapse" data-parent="#aside1" href="#content3" class="collapsed"><p class="first admonition-title">Hardware devices (click to expand) <i class="fas fa-chevron-circle-down"></i></p></a><div id="content3" class="collapse" data-parent="#aside1" style="height: 0px;">`
+
+.. list-table::
+   :widths: 35 45 60 
+   :header-rows: 1
+
+   * - **Device**
+     - **Shortname**
+     - **Description**
+   * - `IBM Q Experience <https://pennylaneqiskit.readthedocs.io/en/latest/devices/ibmq.html>`__
+     - ``"qiskit.ibmq"`` *(must specify a hardware backend)*
+     - IBM Q hardware device, queue based access to IBMQ backends
+   * - `Forest QPUDevice <https://pennylane-forest.readthedocs.io/en/latest/code/qpu.html>`__
+     - ``"forest.qpu"``
+     - Forest QPU device, session based access to Rigetti QPUs
+
+:html:`</div></div>`
+
+:html:`<div class="photonic_list" id="aside1"><a data-toggle="collapse" data-parent="#aside1" href="#content4" class="collapsed"><p class="first admonition-title">Photonic devices (click to expand) <i class="fas fa-chevron-circle-down"></i></p></a><div id="content4" class="collapse" data-parent="#aside1" style="height: 0px;">`
+
+.. list-table::
+   :widths: 35 45 60 
+   :header-rows: 1
+
+   * - **Device**
+     - **Shortname**
+     - **Description**
+   * - :class:`~.DefaultGaussian`
+     - ``"default.gaussian"``
+     - Default gaussian device
+   * - `Strawberry Fields Fock device <https://pennylane-sf.readthedocs.io/en/latest/devices/fock.html>`__
+     - ``"strawberryfields.fock"``
+     - Fock device giving access to the Strawberry Fields Fock state simulator backend
+   * - `Strawberry Fields Gaussian device <https://pennylane-sf.readthedocs.io/en/latest/devices/gaussian.html>`__
+     - ``"strawberryfields.gaussian"``
+     - Gaussian device giving access to the Strawberry Fields Fock state simulator backend
+
+
+:html:`</div></div>`
+
+QNodes
+******
+
+A  quantum  node or ``QNode`` (represented by a subclass to
+:class:`~.BaseQNode`) is an encapsulation of a function :math:`f(x;θ):R^m→R^n`
+that is executed using quantum information processing on a quantum
+device.
+
+Each ``QNode`` represents the quantum circuit by building a
+:class:`~.CircuitGraph` instance, but the way differentiation is done is custom
```

Comment:
This might be confusing to readers, since (I don't believe) CircuitGraph is not mentione elsewhere in the doc

Reply:
included further details